client/extensions/woocommerce/app/settings/shipping/shipping-methods/free-shipping.js

```diff
@@ -1,68 +1,109 @@
 
-		//TODO: use redux state with real data
-		this.state = {
-			everyone: props.everyone,
-			currencySymbolPrefix: '$',
-			currencySymbolSuffix: ''
-		};
-	}
+	const isAdvancedSettings = 'coupon' === requires || 'either' === requires || 'both' === requires;
 
-	renderMinSpendBox() {
-		return (
-			<FormCurrencyInput
-				currencySymbolPrefix={ this.state.currencySymbolPrefix }
-				currencySymbolSuffix={ this.state.currencySymbolSuffix }
-				value={ this.state.minSpend } />
-		);
-	}
+	const renderMinSpendInput = () => (
+		//TODO: remove hardcoded currency settings
+		<FormCurrencyInput
+			currencySymbolPrefix={ '$' }
+			currencySymbolSuffix={ '' }
+			value={ min_amount }
+			disabled={ ! requires }
+			onChange={ onMinAmountChange } />
+	);
 
-	render() {
-		const { translate } = this.props;
+	if ( isAdvancedSettings ) {
+		const renderMinSpendControls = () => {
+			if ( ! requires || 'coupon' === requires ) {
+				return null;
+			}
 
-		const onOptionChange = ( event ) => {
-			this.setState( { everyone: 'everyone' === event.currentTarget.value } );
+			return (
+				<FormFieldSet>
+					<FormLabel>{ translate( 'Minimum order amount:' ) }</FormLabel>
+					{ renderMinSpendInput() }
+				</FormFieldSet>
+			);
 		};
 
 		return (
-			<div className="shipping-methods__method-container">
-				<FormLabel>{ translate( 'Who gets free shipping?' ) }</FormLabel>
-				<div className="shipping-methods__free-shipping-option">
-					<FormRadio value="everyone"
-						checked={ this.state.everyone }
-						onChange={ onOptionChange } />
-					{ translate( 'Everyone!' ) }
-				</div>
-				<div className="shipping-methods__free-shipping-option">
-					<FormRadio value="paying" checked={ ! this.state.everyone } onChange={ onOptionChange } />
-					{ translate( 'Customers that spend {{priceInput/}} or more per order', {
-						components: {
-							priceInput: this.renderMinSpendBox()
-						}
-					} ) }
-				</div>
+			<div>
+				<FormFieldSet>
+					<FormLabel>{ translate( 'Free shipping requires:' ) }</FormLabel>
+					<FormSelect value={ requires } onChange={ onConditionChange }>
+						<option value="">{ translate( 'N/A' ) }</option>
+						<option value="coupon">{ translate( 'A valid free shipping coupon' ) }</option>
+						<option value="min_amount">{ translate( 'A minimum order amount' ) }</option>
+						<option value="either">{ translate( 'A minimum order amount OR a coupon' ) }</option>
+						<option value="both">{ translate( 'A minimum order amount AND a coupon' ) }</option>
+					</FormSelect>
+				</FormFieldSet>
+				{ renderMinSpendControls() }
 			</div>
 		);
 	}
-}
+
+	return (
+		<FormFieldSet>
+			<FormLabel>{ translate( 'Who gets free shipping?' ) }</FormLabel>
+			<div className="shipping-methods__free-shipping-option">
+				<FormRadio
+					value=""
+					checked={ ! requires }
+					onChange={ onConditionChange } />
+				{ translate( 'Everyone!' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Everyone', { context: 'Filter label for posts list'} )` *ES Score: 21*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).